build_tools/index_generation_s3_tar.py

```diff
@@ -0,0 +1,241 @@
+            ul {{ list-style-type: none; padding: 0; }}
+            li {{ margin-bottom: 5px; padding: 10px; background-color: white; border-radius: 5px; box-shadow: 0 0 5px rgba(0,0,0,0.1); }}
+            a {{ text-decoration: none; color: #0056b3; word-break: break-all; }}
+            a:hover {{ color: #003d82; }}
+            .controls {{ display: flex; gap: 12px; align-items: center; flex-wrap: wrap; }}
+            label {{ font-weight: bold; }}
+        </style>
+        <script>
+            const files = {files_js_array};
+            function applyFilter(fileList, filter) {{
+                if (filter === 'all') return fileList;
+                return fileList.filter(file => file.name.includes(filter));
+            }}
+            function renderFiles(fileList) {{
+                const ul = document.getElementById('fileList');
+                ul.innerHTML = '';
+                fileList.forEach(file => {{
+                    const li = document.createElement('li');
+                    const href = encodeURIComponent(file.name).replace(/%2F/g, '/');
+                    li.innerHTML = `<a href="${{href}}" target="_blank" rel="noopener noreferrer">${{file.name}}</a>`;
+                    ul.appendChild(li);
+                }});
+            }}
+            function updateDisplay() {{
+                const order = document.getElementById('sortOrder').value;
+                const filter = document.getElementById('filter').value;
+                let sortedFiles = [...files].sort((a, b) => {{
+                    return (order === 'desc') ? b.mtime - a.mtime : a.mtime - b.mtime;
+                }});
+                sortedFiles = applyFilter(sortedFiles, filter);
+                renderFiles(sortedFiles);
+            }}
+            document.addEventListener('DOMContentLoaded', function() {{
+                updateDisplay();
+                document.getElementById('sortOrder').addEventListener('change', updateDisplay);
+                document.getElementById('filter').addEventListener('change', updateDisplay);
+            }});
+        </script>
+    </head>
+    <body>
+        <h1>{page_title}</h1>
+        <div class="controls">
+            <label for="sortOrder">Sort by:</label>
+            <select id="sortOrder">
+                <option value="desc">Last Updated (Recent to Old)</option>
+                <option value="asc">First Updated (Old to Recent)</option>
+            </select>
+            <label for="filter">Filter by:</label>
+            <select id="filter">
+                <option value="all">All</option>
+                {gpu_families_options}
+            </select>
+        </div>
+        <ul id="fileList"></ul>
+    </body>
+    </html>
+    """
+
+    # Write locally
+    local_path = "index.html"
+    with open(local_path, "w", encoding="utf-8") as f:
+        f.write(html_content)
+
+    message = f"index.html generated successfully for bucket '{bucket_name}'. File saved as {local_path}"
```

Comment:
can we create a better log message? Maybe 

```suggestion
    message = f"Index page generated for bucket '{bucket_name}'. File saved as {local_path}"
```